### PR TITLE
Move Delete TableWriter TableFinish node to SPI

### PR DESCRIPTION
**Pull request by feilong-liu**
## Description
We are adding a new connector optimizer, however the connector optimizer can only access a limited set of plan nodes which are in the SPI directory. In this PR, I move the DeleteNode, TableWriter and TableFinish node to SPI so that they can be used in collector optimizer.

Most part of the change is refactoring the import path for these nodes. Meaningful change is to get rid of functions not available in SPI, mainly Guava libraries, and replace them with java builtin collection methods.

## Motivation and Context
To make DeleteNode, TableWriter and TableFinish nodes available in connector optimizer

## Impact
Move Delete TableWriter TableFinish node to SPI

## Test Plan
Existing unit tests

## Contributor checklist

- [ ] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [ ] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [ ] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [ ] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [ ] Adequate tests were added if applicable.
- [ ] CI passed.

## Release Notes
Please follow [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) and fill in the release notes below.

```
== RELEASE NOTES ==

General Changes
* Add Delete TableWriter TableFinish node to SPI :pr:`24088`
```

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/sql/TemporaryTableUtil.java [hunk 1]
```diff
@@ -384,4 +389,59 @@ public static TableFinishNode createTemporaryTableWriteWithExchanges(
                 enableStatsCollectionForTemporaryTable ? Optional.of(aggregations.getFinalAggregation()) : Optional.empty(),
                 enableStatsCollectionForTemporaryTable ? Optional.of(statisticsResult.getDescriptor()) : Optional.empty());
     }
+
+    public static StatisticAggregations.Parts splitIntoPartialAndFinal(StatisticAggregations statisticAggregations, VariableAllocator variableAllocator, FunctionAndTypeManager functionAndTypeManager)
```
Comment (feilong-liu) [inline presto-main/src/main/java/com/facebook/presto/sql/TemporaryTableUtil.java:393]: Moved from StatisticAggregations, as it's moved to SPI which has no access to FunctionAndTypeManager

File: presto-main/src/main/java/com/facebook/presto/sql/planner/plan/StatisticAggregations.java [hunk 2]
```diff
@@ -1,170 +0,0 @@
...
-import com.google.common.collect.ImmutableMap;
-
-import java.util.List;
-import java.util.Map;
-import java.util.Optional;
-
-import static com.google.common.base.Preconditions.checkArgument;
-import static java.util.Objects.requireNonNull;
-
-public class StatisticAggregations
```
Comment (feilong-liu) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/plan/StatisticAggregations.java:35]: Moved to SPI

**Review by aditi-pandit — APPROVED** [review]
For Native engine changes.

**Comment by steveburnett** [comment]
Thanks for the release note entry! Please revise it to follow the [Order of changes](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines#order-of-changes) in the Release Note Guidelines to begin the line with _Fix_, _Improve_, and so on.

**Review by rschlussel — APPROVED** [review]

**Pull request merged**